fabric-screen-handler-api-v1/src/main/java/net/fabricmc/fabric/api/screenhandler/v1/SlotWithClickAction.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.screenhandler.v1;
+
+import net.minecraft.entity.player.PlayerInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ClickType;
+
+/**
+ * An interface for slots that consume a click action.
+ * Slot click actions are called before item click actions, and returning true will neglect to have them called whatsoever.
+ */
+public interface SlotWithClickAction {
```

Comment:
I prefer to name this `ClickAwareSlot`

Reply:
I think `SlotWithClickAction` is better as it represents the slot having an action when it's clicked.

Reply:
ClickAwareSlot is shorter, but less descriptive. All slots are clickable, but only these slots have a special click action.

Reply:
`SlotWithClickAction` sounds better and easier to understand. Shorter names aren't always the best when you lose important details

Reply:
Then `ClickCallbackSlot` imo is better. All slot clicks have click action objects assigned, dont' they?

Reply:
This isn't really what I think of when I think of a callback, but also, slots by themselves *do not* have a click action, screen handlers mutate them manually, the same way items didn't have any until 1.17. I think calling what happens when a click is handled differently that standard a `ClickAction` makes more sense by far than a `ClickCallback`.

Reply:
every slot can be clicked, not every slot has a click action. SlotClickCallback implies you're passing a seperate callback object elsewhere, and ClickCallbackSlot is just more confusing and longer.

Reply:
But this is a callback. You are using the slot object effectively as a callback object and how this method is called is like a callback. `XxWithYy` are generally confusing names for our naming style, just like `CallbackInfoReturnable` is a bad name because it is an info than a returnable.

In addition, vanilla has a click action for text. This use of click action will just be like what juuxel said, increase confusion.

Reply:
at this point you're being contrary for the sake of being contrary. all other reviews agree with the existing name and find it to be the best suggested name.

Reply:
`ClickActionSlot` and be done with it. ;)

Reply:
> at this point you're being contrary for the sake of being contrary.

Lol what, this is just that typical hypocrisy where you say you allow freedom of speech but enforce people to think alike. So much for arguments! When you cannot debate, you eliminate the opposition!

I still hold that names with `With`, like `ScreenWithHandler`, are confusing, just like speaking in passive voice than the active voice. My emphasis is that it's `ClickAware` and it serves as a `ClickCallback` so it comes first in name. Yours just don't align with the suffix naming style.

Reply:
I don't like `With`, but I don't like `Aware` either. We can just remove `With` and go for `ClickActionSlot` imo.

Reply:
It's a slot with a click action, it's not a click action slot, I do agree that `With` is generally not great, but when something is defined as being one thing with another thing, that naming scheme is at the very least applicable, and in my opinion ideal.

Reply:
if anyone asks me where to put their bikes I'll link them to this thread

Reply:
`ClickActionSlot` -> `Slot`, but also has a `ClickAction`. We may want to emphasize somehow that this click action happens before `Item#onClick`.

Reply:
> Lol what, this is just that typical hypocrisy where you say you allow freedom of speech but enforce people to think alike. So much for arguments! When you cannot debate, you eliminate the opposition!

100% agreement never happens for any debate. Being in the minority does not mean your "freedom of speech" is being suppressed, it simply means most people don't believe your point. Don't try to bring that crap into a technical discussion, it's obviously in bad faith.

This just reads to me as "if I don't get it my way or people don't agree with me, then you're suppressing my right to debate", which is patently false.

Honestly, Fabric API has a huge reputation for bikeshedding, to the point where I have spoken to people that are afraid to even contribute because minor, inconsequential pieces of their PR will be nitpicked and bikeshedded to death. And from my brief experience contributing and interacting with some reviewers, that's true. That doesn't sound healthy for a project, especially since there's an example in the project next door of a supposedly unhealthy contribution culture.

Reply:
> ClickActionSlot -> Slot, but also has a ClickAction. We may want to emphasize somehow that this click action happens before Item#onClick.
Documentation for both the interface and the method specifies that this is called beforehand, I think it would be cluttersome to put in the method name somehow.

Reply:
> 
> 
> I don't like `With`, but I don't like `Aware` either. We can just remove `With` and go for `ClickActionSlot` imo.

This is a good one. :+1:

Reply:
> 
> 
> > Lol what, this is just that typical hypocrisy where you say you allow freedom of speech but enforce people to think alike. So much for arguments! When you cannot debate, you eliminate the opposition!
> 
> 100% agreement ~never~ rarely happens for any debate. Being in the minority does not mean your "freedom of speech" is being suppressed, it simply means most people don't believe your point. Don't try to bring that crap into a technical discussion, it's obviously in bad faith.
> 
> This just reads to me as "if I don't get it my way or people don't agree with me, then you're suppressing my right to debate", which is patently false.

Another TLDR, huh? Also in your quote above, you missed this:
` you're being contrary for the sake of being contrary.`
What is the purpose of this sentence? It clearly asks me to stop arguing.

Anyways, I am bikeshedding in my domain, i.e. suggesting intuitive and efficient names for things. That's why I send pull requests to yarn. In addition, I am not requesting changes to this pr for this name; the rest of the pr is great.

Reply:
Most people seem to prefer `SlotWithClickAction` to any alternatives suggested, `ClickActionSlot` isn't very clear what exactly it is either. Is it a slot with a click action? Or an event for clicking an action slot? A slot that *is* a click action? Sometimes with brevity you lose clarity, and I think this is a situation where that is the case.

Reply:
I prefer `SlotClickListener` or `SlotClickHandler`.

Reply:
Let's go for `SlotWithClickAction`, it should be clear enough. `Listener` or `Handler` is too vague because it doesn't convey that the slot itself is handling the click action.

Reply:
Sometimes you need to use longer names to clearly specify what the class does. Hence why we have `LivingEntityFeatureRendererRegistrationCallback`.

I think the name is fine. For making it more obvious what to do with the interface, I would add some documentation along the lines of "To allow for enhanced click action handling, implement this interface onto a Slot".

Reply:
> For making it more obvious what to do with the interface, I would add some documentation along the lines of "To allow for enhanced click action handling, implement this interface onto a Slot".

is `An interface for {@link net.minecraft.screen.slot.Slot}s that consume a click action.` not ideal? Would it be better to use the phrase "enhanced click action handling" or something similar to this effect?

Reply:
The current comment is sufficient, but elaborating on what you do with the interface explicitly (implement the interface) is much more clear to a new user.

Reply:
I can't close this chain because it has been about more than just docs, but I've added a bit of elaboration to that tune.